### ISSUE TITLE
CU - Make appointments data slightly more sensical

### ISSUE THREAD

**Issue by rbontrager** [department-of-veterans-affairs/va-mobile-app#8931]
## Proposed Change

When doing the detox work for appts componentization I prioritized making sure that the correct fields were filled in and the demo appts were acting like they would have in staging.  This means that sometimes we have location data that doesn't line up with the facility name etc.  This ticket it to continue cleaning up appts in demo mode and make them slightly more sensical where possible.

## Acceptance Criteria
 

- Appointments address/phone number should line up with the location name
- Appointments address data should line up with the clinic info
- Pending date should be around the date on the appointments list

## Checklist
- [x] Add the relevant team label (Health, global, design system, API, Qa and Release etc.)
- [x] Attach to ticket to the relevant Team Tech Debt Epic epic (old frontend engineering epic is no longer in use as each team is managing their code's technical debt and code upkeep work)

**Comment by Sparowhawk** [comment]
@rbontrager @dumathane I wonder if it makes sense to create the various appts on a staging user and then we just copy and paste the data over to demo mode and then make whatever changes we need to do for special scenarios(null data, setting dates to be X days in the future or past from current date, etc...)

**Comment by timwright12** [comment]
@rbontrager can we get some acceptance criteria on this ticket?

**Comment by jmarchi** [comment]
@BeccaTupaj Yes, that is my read on the description/AC. I'm going through looking for inconsistencies

**Comment by jmarchi** [comment]
Findings included in ticket https://github.com/department-of-veterans-affairs/va-mobile-app/issues/11947 cc @BeccaTupaj
